src/http/common.cr

```diff
@@ -95,9 +109,28 @@ module HTTP
       true
     end
   end
+
+  def self.parse_time(time_str : String)
+    patterns = ["%a, %d %b %Y %H:%M:%S %z", "%A, %d-%b-%y %H:%M:%S %z", "%a %b %e %H:%M:%S %Y"]
+    patterns.each do |pattern|
+      begin
+        return Time.parse(time_str, pattern)
+      rescue
+        # TODO: do something here?
+      end
+    end
+
+    return nil
+  end
```

Comment:
Well, there appears to be no non-raising API for this, so rescue the specific exception (`TimeFormat::Error`) and we should be good.